src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -547,10 +547,26 @@ withNewArrayUnsafe count f = do
 -- | Write an input stream of (index, value) pairs to an array. Throws an
 -- error if any index is out of bounds.
 --
--- /Unimplemented/
+-- /Pre-release/
 {-# INLINE putIndices #-}
-putIndices :: Array a -> Fold m (Int, a) ()
-putIndices = undefined
+putIndices :: forall m a. (MonadIO m, Storable a)
+    => Array a -> Fold m (Int, a) ()
+putIndices Array{..} = Fold step initial extract
```

Comment:
This would just be `Fold.drainBy (uncurry (putIndex arr))`. Should not make any difference to perf as each element is anyway independently modified.